### PR TITLE
Add AdamW tutorial.

### PR DESCRIPTION
**Pull request by us**
Related issue: #336

### CODE REVIEW

**Comment by us** [comment]
I worked on colab but I couldn't find the save notebook with outputs. How can I fix this problem?

**Comment by seanpmorgan** [comment]
> I worked on colab but I couldn't find the save notebook with outputs. How can I fix this problem?

So after running the notebook if you select "Save a copy in GitHub" the full notebook with results should save:
https://github.com/tensorflow/addons/tree/master/docs/tutorials#how-to-contribute 

Please let us know if that documentation is out of date or isn't working for some reason.

**Comment by us** [comment]
Thanks for answering Sean, I missed it on doc, I'm doing it right now.

**Comment by us** [comment]
@seanpmorgan I fixed it. The first commit is a mistake, so should I delete this commit?

**Comment by us** [comment]
@seanpmorgan should I change or add anything?

File: docs/tutorials/optimizers_adamw.ipynb [hunk 1]
```diff
@@ -0,0 +1,692 @@
+{
```
Comment (seanpmorgan) [inline docs/tutorials/optimizers_adamw.ipynb]: <p>I worry  that saying "all the popular Deep Learning frameworks (Tensorflow, Pytorch) have implemented Adam with weight decay wrong." will become out of date (if it is not already). Could you remove that comment please.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/1217/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (seanpmorgan) [inline docs/tutorials/optimizers_adamw.ipynb]: <p>Can you remove the output of this cell. Displaying the results in graphical form below is sufficient.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/1217/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (seanpmorgan) [inline docs/tutorials/optimizers_adamw.ipynb]: <p>ditto</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/1217/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (seanpmorgan) [inline docs/tutorials/optimizers_adamw.ipynb]: <p>Results don't seem to be showing the benefit of the optimizer. Do we think this is do to the number of epochs? Perhaps try training for longer to see. If not and we believe the implementation is correct I'm still good with merging this tutorial as an guide to use it.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/1217/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (us) [reply docs/tutorials/optimizers_adamw.ipynb]: If TensorFlow's adam use weight decay automatically then we should disable but how can I do it?
The difference between `adam` and `adamw` becomes evident at the end of the graphic. So I will try to increase epoch size.

Comment (us) [reply docs/tutorials/optimizers_adamw.ipynb]: Now, I think the last commit showed the benefit of AdamW better.

**Comment by MHStadler** [comment]
The AdamW paper comes with 3 additional important insights for the use of Adam, beyond just the decoupled weight decay:
 - Despite being an adaptive method, it can benefit from a LR schedule (a cosine schedule in the case of their paper - wd schedule is now fixed: https://github.com/tensorflow/addons/pull/2029)
 - L2 regularization does not work as well for Adam as it does for SGD
 - WD should be normalized to be more stable across short/long runs (see Appendix B.1)

So, if you're still looking to improve the graphs, I think this type of comparison would better illustrate the potential of AdamW:
 - SGD with L2
 - Adam with L2
 - AdamW with cosine decay 

What you'd hope to see is that AdamW with the changes from the paper closes the gap between traditional Adam with L2 penalty & SGD with momentum & L2 penalty

**Comment by bhack** [comment]
AdamW now is Keras:
https://github.com/keras-team/keras/blob/master/keras/optimizers/optimizer_experimental/adamw.py

You can add this tutorial at https://github.com/keras-team/keras-io

Thanks
